### ISSUE TITLE
false positive for no-unused-vars

### ISSUE THREAD

**Issue by psabharwal123** [eslint/eslint#12242]
**Tell us about your environment**

* **ESLint Version:** 6.3.0
* **Node Version:** 10.16.3
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
	extends: 'eslint-config-pogo',
	rules: {
		'promise/param-names': 'off'
	},
	env: {
		jest: true
	}
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
        if (this.props.data && this.props.tileData) {
            for (const key in this.props.tileData) {
                tiles.push(
                    Object.assign({}, this.props.tileData[key], {
                        id: parseInt(key.replace(/\D/g, ''), 10),
                        size: (this.props.tileData[key].size || '').toLowerCase()
                    })
                );
            }
        }
```
and
```js
            if (task.errors) {
                for (const name in task.errors) {
                    if (errors[name]) {
                        errors[name] = [...errors[name], ...task.errors[name]];
                    } else {
                        errors[name] = task.errors[name];
                    }
                }
            }
```

```bash
eslint --ext js,jsx src library
```

**What did you expect to happen?**
no errors

**What actually happened? Please include the actual, raw output from ESLint.**
eslint no-unused-vars errors

**Are you willing to submit a pull request to fix this bug?**
no

**Comment by mysticatea** [comment]
Your config is using `babel-eslint` (not default parser). Please ensure to use the latest `babel-eslint`.

I'm closing as a duplicate of #12117.

**Comment by psabharwal123** [comment]
babel-eslint does not even exist in package.json

**Comment by psabharwal123** [comment]
and no it is not a duplicate, #12117 wasn't fixed completely - https://github.com/eslint/eslint/issues/12117#issuecomment-525133544

**Comment by mysticatea** [comment]
You have `extends: 'eslint-config-pogo'` and `eslint-config-pogo` depends on `babel-eslint`.
